Update install script to install default plugins and templates (#643)
fermyon/developer: content/spin/quickstart.md

{{ startTab "Linux"}}

> Download the `spin` binary along with a starter set pf templates and plugins using the `install.sh` script hosted on this site:

Reviewer: ```suggestion
Download the `spin` binary along with a starter set of templates and plugins using the `install.sh` script hosted on this site:
```

> Download the `spin` binary along with [default templates and plugins](install.md#installing-default-templates-and-plugins) using the `install.sh` script hosted on this site:

Reviewer 2: Again let's remove this link.

Reviewer: Thanks Kate.

{{ startTab "macOS"}}

> Download the `spin` binary along with language templates and plugins using the `install.sh` script hosted on this site:

Reviewer: ```suggestion
Download the `spin` binary along with a starter set of templates and plugins using the `install.sh` script hosted on this site:
```

> Download the `spin` binary along with language templates and plugins using the `install.sh` script hosted on this site:

Reviewer 2: ```suggestion
Download the `spin` binary along with a starter set of language templates and plugins using the `install.sh` script hosted on this site:
```

(for consistency)

Reviewer: This is updated in the latest souce.

Unzip the binary release and place the `spin.exe` in your system path.

Next, follow the steps to install the [default templates and plugins](#installing-default-templates-and-plugins).

{{ blockEnd }}
{{ blockEnd }}

[See more options for installing Spin.](install)

## Install the Prerequisites

> > If you installed Spin with the install script, you can skip these steps, as the script installs the [default templates and plugins](install.md#installing-default-templates-and-plugins). List installed templates and plugins by running `spin templates list` and `spin plugins list`.

Reviewer 2: The conditional stuff and big bunch of stuff to skip complicates the quick start a bit... I wonder if we should now position this under the Windows section because for Linux and Mac it is already done, but I doubt we can do language tabs within OS tabs.

This gets even nastier in the next section, "Install the Tools," because now for Rust and Go we still need you to install the relevant target/compiler, but for JS and Python we no longer need you to install the plugin, _unless_ you are on Windows in which case...

It would be really good to keep the quick start on rails, with no "oh now I need to skim and figure out the next step that's relevant to me."  But I am not sure what our options are.

Reviewer 3: I think we can simply remove this. Worst case, is that people will run the commands with an update flag, and everything will be fine :-)

Author: I agree with removing the warning, since it is a good walk through of what the template and plugin systems are and how to engage with them

Author: I agree with removing the warning, since it is a good walk through of what the template and plugin systems are and how to engage with them

Reviewer: Thanks Kate, I see this is removed now.

### Install a Template

The quickest and most convenient way to start a new application is to use a Spin template.

> Let's install the templates for your preferred language.

Reviewer 2: nit: suggest combining this with the previous paragraph, as it was before (I don't feel strongly about it, just the change feels a bit superfluous now that there's no longer the extra words about whether or not you need to do it)
